enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,348 @@
+---
+title: heterogeneous-architecture-clusters
+authors:
+  - "@Prashanth684"
+reviewers:
+  - "@bparees"
+  - "@derekwaynecarr"
+  - "@jupierce"
+  - "@sdodson"
+  - "@wking"
+  - "@LalatenduMohanty - upgrades"
+  - "@sosiouxme - payload build and release"
+  - "@enxebre - hypershift"
+  - "@dmage - imagestream api changes"
+  - "@kdoberst - console filtering changes"
+  - "@adambkaplan - Openshift Builds"
+  - "@perdasilva - OLM changes"
+  - "@mrunalp - MCO changes"
+approvers:
+  - "@bparees"
+  - "@derekwaynecarr"
+creation-date: 2022-01-17
+last-updated: 2022-01-17
+status: implementable
+---
+
+# heterogeneous-architecture-cluster-support
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for tech preview and GA
+- [ ] User-facing documentation is created in [Openshift-docs](https://github.com/Openshift/Openshift-docs/)
+
+## Summary
+
+This enhancement describes the [plan][OCPPLAN-4577] to support provisioning and upgrades of heterogeneous architecture Openshift clusters.
+A heterogeneous cluster is a cluster with nodes running on different architectures with a homogeneous control plane and worker node pools
+supporting any combination of architectures (eg: x86 control plane + ARM worker nodes). The node pools within the cluster are homogeneous.
+This document describes the support needed for heterogeneous architecture clusters in phases. There will be three major phases with basic
+support in phase 1 which is targeted for 4.11 and the other phases which address gaps beyond the 4.11 time frame. The targeted platforms
+in the initial phases are Azure and AWS with support for x86 and ARM architectures.
+
+![alt heterogeneous cluster](heterogeneous-cluster.png)
+
+## Motivation
+
+- Users can take advantage of cost savings on architectures like ARM.
+- Users can colocate applications that require different architectures rather than needing multiple clusters and thus avoid cross cluster
+latency and connectivity failures.
+
+### Goals
+
+- Provide a way to install and upgrade clusters with varied architecture node pools with the ability to autoscale heterogeneous machinesets.
+- Support provisioning heterogeneous architecture clusters in Hypershift.
+- Generate a manifestlist based payload that contains manifestlist references for the 4 architectures OpenShift supports (x86, ARM, Power and Z).
```

Comment:
Is the goal to have different architectures among worker nodes as well? For example if there are 100 worker nodes, can be 25 nodes of x86, 25 nodes of ARM, 25 nodes of power?

Reply:
yes....the goal is not just "control plane nodes are one arch, and workers are a different arch", the goal is "workers can have a mix of arches"

Reply:
changed the wording in the summary section to reflect this.